src/index.tsx

```diff
@@ -152,9 +156,34 @@ const MetaDiagram = forwardRef(
       eventWillFire: preCallback,
     });
 
+    const clearSelection = () => {
+      engine.getModel().clearSelection();
+    };
+
+    // update state selection state
+    const updateSelection = (id: string) => {
+      if (
+        id.toLowerCase().startsWith(DefaultSidebarNodeTypes.SELECT) &&
+        !Boolean(state.isSelection)
+      ) {
+        state.isSelection = true;
+      } else if (
+        id.toLowerCase().startsWith('select') &&
```

Comment:
I think you may want to use DefaultSidebarNodeTypes.SELECT instead of 'select':

And maybe extract the common term outside so you don't repeat yourself. Something like:

const updateSelection = (id: string) => {
  const startsWithSelect = id.toLowerCase().startsWith('DefaultSidebarNodeTypes.SELECT');
 ...
};